### PR TITLE
feat(crons): Display service incidents on timeline

### PR DESCRIPTION
**Pull request by evanpurkhiser**

### CODE REVIEW

File: static/app/views/monitors/components/timeline/gridLines.tsx [hunk 1]
```diff
@@ -23,6 +24,10 @@ interface Props {
    * Enable the timeline cursor
    */
   showCursor?: boolean;
+  /**
+   * Render sentry service incidents as an overlay
+   */
+  showIncidents?: boolean;
```
Comment (davidenwang) [inline static/app/views/monitors/components/timeline/gridLines.tsx:30]: hmm are there instances where we actually don't want to show the incidents besides `isLoading`? I wonder if we should just have a `isLoading` here which disables/enables both `showCursor` and `showIncidents`

File: static/app/views/monitors/components/timeline/serviceIncidents.tsx [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+import type {TimeWindowConfig} from './types';
+
+interface CronServiceIncidentsProps {
+  timeWindowConfig: TimeWindowConfig;
+}
+
+interface GetTimeWindowConfigOptions {
+  inc: StatuspageIncident;
+  /**
+   * Clamo the incident start and end times to be the timeWindowConfig start
```
Comment (davidenwang) [inline static/app/views/monitors/components/timeline/serviceIncidents.tsx:24]: nit: Clamp*

**Review by davidenwang — APPROVED** [review]
this is sweet

File: static/app/views/monitors/components/timeline/gridLines.tsx (hunk 1 above)
Comment (evanpurkhiser) [reply static/app/views/monitors/components/timeline/gridLines.tsx:30]: yeah we don't show it on the visualization either

Comment (davidenwang) [reply static/app/views/monitors/components/timeline/gridLines.tsx:30]: ahh i see

**Pull request merged**